server/controllers/index.js

```diff
@@ -38,22 +25,52 @@ exports.getNearbyBuildings = asyncHandler( async (req, res, next) => {
       }
     },
     workspaceId: { $ne: origin.workspaceId, $lte: 100, $gte: 1 }
-  });
+  })
+    .then(() => {
+      res.status(200).json({ origin, nearbyWorkspaces });
+    })
+    .catch(() => {
+      console.log('No nearby workspaces generated');
+      res.status(200).json({ origin })
+    })
+})
 
-  res.status(200).json({ origin, nearbyWorkspaces });
+workspaceRouter.post('/:workspaceId', async (req, res) => {
+  const origin = await WorkspaceLocation.create( { ...req.body })
+  res.status(200).json({ origin });
+})
 
+workspaceRouter.put('/:workspaceId', async (req, res) => {
+  const { workspaceId } = req.params;
+  const origin = await WorkspaceLocation.findOneAndUpdate( { workspaceId }, { ...req.body }, { new: true });
+  checkRecord(origin);
+  res.status(200).json({ origin });
 });
 
+workspaceRouter.delete('/:workspaceId', async (req, res) => {
+  const { workspaceId } = req.params;
+  const origin = await WorkspaceLocation.findOneAndRemove( { workspaceId });
+  checkRecord(origin);
+  res.status(200).json({ origin });
+});
+
+
 // return an error for non existing routes
 // ALL @ *
-exports.notFound = asyncHandler( async (req, res, next) => {
+const notFound = asyncHandler( async (req, res, next) => {
   const err = new Error('Endpoint does not exist');
   err.status = 404;
   throw err;
 })
 
 // return an simple response when errors thrown in endpoints
-exports.errorHandler = (err, req, res, next) => {
+const errorHandler = (err, req, res, next) => {
   console.log(err);
   res.status(err.status || 500).send({ success: false, status: err.status || 500, message: err.message});
 };
+
+module.exports = {
+  workspaceRouter,
+  notFound,
+  errorHandler,
+};
```

Comment:
Yeah, thanks for reiterating that. I usually use the try/catch convention, but I wanted to try out what I inherited because I had never seen it before. I did notice issues with it when I was testing, though, because errors were not being caught correctly.